modules/openapi-generator/src/main/java/org/openapitools/codegen/IJsonSchemaValidationProperties.java

```diff
@@ -226,11 +226,8 @@ public interface IJsonSchemaValidationProperties {
      * @param p the schema which contains the type info
      */
     default void setTypeProperties(Schema p) {
-        if (ModelUtils.isTypeObjectSchema(p)) {
-            setIsMap(true);
-            if (ModelUtils.isModelWithPropertiesOnly(p)) {
+        if (ModelUtils.isModelWithPropertiesOnly(p)) {
                 setIsModel(true);
```

Comment:
You may be able to remove these if statements like `setIsModel(ModelUtils.isModelWithPropertiesOnly(p))`

Reply:
Thanks for suggestion. Will take that into consideration.

Personally I prefer the current flow as some checks should be done last as the check order matters.

Reply:
Not quite sure I understand what you mean. Maybe I wasn't clear.

```java
default void setTypeProperties(Schema p) {
    setIsModel(ModelUtils.isModelWithPropertiesOnly(p));
    setIsArray(ModelUtils.isArraySchema(p));
    setIsNull(ModelUtils.isFileSchema(p) && !ModelUtils.isStringSchema(p));
    setIsAnyType(ModelUtils.isAnyType(p));
    setIsMap(ModelUtils.isTypeObjectSchema(p));
}
```

The order does not change. Thanks for considering.

Reply:
The `ModelUtils.isTypeObjectSchema(p)` check should perform (with an `else`) only if none of the previous checks match so that we can avoid `isModel`, `isMap` set to true at the same time for the same schema.